source/slang/slang-emit-cpp.cpp

```diff
@@ -1591,6 +1609,83 @@ void CPPSourceEmitter::emitWitnessTable(IRWitnessTable* witnessTable)
     pendingWitnessTableDefinitions.add(witnessTable);
 }
 
+void CPPSourceEmitter::_emitWitnessTableWrappers()
```

Comment:
This logic is obviously a good warning sign to indicate that we eventually want to synthesize these wrappers at the IR level, rather than it being deeply entwined with how we emit code.